build.rs

```diff
@@ -1,21 +1,35 @@
+use std::io;
 use std::process::Command;
 
 const GIT_HASH_ENV_VAR: &str = "GIT_SHA_SHORT";
 
 fn main() {
-    //Build command to retrieve the short git commit hash
-    let git_process_output = Command::new("git")
+    match get_git_commit_hash() {
+        // If the git commit hash is retrieved successfully, set the environment variable
+        Ok(git_hash) => {
+            println!("cargo:rustc-env={GIT_HASH_ENV_VAR}={git_hash}");
+        }
+        // If there's an error retrieving the git commit hash, print a note and set the environment variable to "unknown"
+        Err(e) => {
+            eprintln!("Note: Failed to get git commit hash: {}", e);
```

Comment:
I think `cargo:warning=MESSAGE` is meant to be used in build scripts for printing output
```suggestion
            println!("cargo:warning=Note: Failed to get git commit hash: {}", e);
```